script.embuary.info/resources/lib/helper.py

```diff
@@ -0,0 +1,235 @@
+
+    try:
+        xbmc.log(msg=message.encode('utf-8'), level=loglevel) # Python 2
+    except TypeError:
+        xbmc.log(msg=message, level=loglevel)
+
+
+def get_cache(key):
+    if CACHE_ENABLED:
+        return CACHE.get(key)
+
+
+def write_cache(value,data):
+    if CACHE_ENABLED:
+        CACHE.set(value,data,expiration=datetime.timedelta(days=14))
+
+
+def format_currency(integer):
+    try:
+        integer = int(integer)
+        if integer < 1:
+            raise Exception
+
+        return '{:,.0f}'.format(integer)
+
+    except Exception:
+        return ''
+
+
+def sort_dict(items,key,reverse=False):
+    try:
+        return sorted(items,key=operator.itemgetter(key),reverse=reverse)
+
+    except KeyError:
+        for item in items:
+            if not item.get(key):
+                item[key] = '0'
+
+        return sorted(items,key=operator.itemgetter(key),reverse=reverse)
+
+
+def remove_quotes(label):
+    if not label:
+        return ''
+
+    if label.startswith("'") and label.endswith("'") and len(label) > 2:
+        label = label[1:-1]
+        if label.startswith('"') and label.endswith('"') and len(label) > 2:
+            label = label[1:-1]
+
+    return label
+
+
+def get_date(date_time):
+    date_time_obj = datetime.datetime.strptime(date_time, '%Y-%m-%d %H:%M:%S')
+    date_obj = date_time_obj.date()
+
+    return date_obj
+
+
+def execute(cmd):
+    xbmc.executebuiltin(cmd, DEBUG)
+
+
+def visible(condition):
+    return xbmc.getCondVisibility(condition)
+
+
+def busydialog(close=False):
+    if not close and not visible('Window.IsVisible(busydialognocancel)'):
+        execute('ActivateWindow(busydialognocancel)')
+    elif close:
+        execute('Dialog.Close(busydialognocancel)')
+
+
+def winprop(key, value=None, clear=False, window_id=10000):
+    window = xbmcgui.Window(window_id)
+
+    if clear:
+        window.clearProperty(key.replace('.json', '').replace('.bool', ''))
+
+    elif value is not None:
+
+        if key.endswith('.json'):
+            key = key.replace('.json', '')
+            value = json.dumps(value)
+
+        elif key.endswith('.bool'):
+            key = key.replace('.bool', '')
+            value = 'true' if value else 'false'
+
+        window.setProperty(key, value)
+
+    else:
+        result = window.getProperty(key.replace('.json', '').replace('.bool', ''))
+
+        if result:
+            if key.endswith('.json'):
+                result = json.loads(result)
+            elif key.endswith('.bool'):
+                result = result in ('true', '1')
+
+        return result
+
+
+def date_format(value,date='short'):
+    try:
+        try:
+            date_time = arrow.get(value, 'YYYY-MM-DD')
+        except Exception:
+            date_time = arrow.get(value, 'DD MMM YYYY')
+
+        value = date_time.strftime(xbmc.getRegion('date%s' % date))
+
+    except Exception:
+        pass
+
+    return value
+
+
+def get_bool(value,string='true'):
```

Comment:
This doesn't make sense. Kodi python API now has methods to get settings as bool. Use those instead

Reply:
I use this function to compare other this too and return false or true. just a little helper. I don't use it to get kode settings.

Reply:
Ok